components/claim/components/email_to_ldap.tsx

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useRef} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {AuthChangeResponse} from '@mattermost/types/users';
+
+import {emailToLdap} from 'actions/admin_actions.jsx';
+
+import {localizeMessage} from 'utils/utils';
+import {t} from 'utils/i18n';
+import {ClaimErrors} from 'utils/constants';
+
+import LoginMfa from 'components/login/login_mfa';
+import LocalizedInput from 'components/localized_input/localized_input';
+
+type Props = {
+    email: string | null;
+    siteName?: string;
+    ldapLoginFieldName?: string;
+}
+
+export type SubmitOptions = {
+    loginId: string;
+    password: string;
+    token?: string;
+    ldapIdParam?: string;
+    ldapPasswordParam?: string;
+}
+
+const EmailToLDAP = ({email, siteName, ldapLoginFieldName}: Props) => {
+    const emailPasswordInput = useRef<HTMLInputElement>(null);
+    const ldapIdInput = useRef<HTMLInputElement>(null);
+    const ldapPasswordInput = useRef<HTMLInputElement>(null);
+
+    const [password, setPassword] = React.useState('');
+    const [ldapId, setLdapId] = React.useState<string | null>(null);
+    const [ldapPassword, setLdapPassword] = React.useState<string | null>(null);
```

Comment:
we could also remove the `null` here and just use an empty string

Reply:
Done, thanks